later/lib/presentation/cubit/home_cubit/home_state.dart

```diff
@@ -0,0 +1,13 @@
+abstract class HomeState {}
+
+class HomeEmpty extends HomeState {}
+
+class HomeLoading extends HomeState {}
+
+class HomeLoaded extends HomeState {
+  HomeLoaded(this.recordsList);
+
+  final Stream<List<RecordModel>> recordsList;
+}
+
+class HomeError extends HomeState {}
```

Comment:
Use `freezed` for this. In the state you should have these props:
- bool isLoading
- Exception error
- List<RecordModel> records

When you emit new states, you can do this:
```dart
emit(state.copyWith(isLoading: true))
```